src/app/reporting/Engine.cpp

```diff
@@ -185,13 +190,15 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeReportIBs(ReportDataMessage::Bu
                              "Error retrieving data from clusterId: " ChipLogFormatMEI ", err = %" CHIP_ERROR_FORMAT,
                              ChipLogValueMEI(pathForRetrieval.mClusterId), err.Format());
 
-                // If error is not CHIP_ERROR_BUFFER_TOO_SMALL and is not CHIP_ERROR_NO_MEMORY, rollback and encode status.
+                // If error is not an "out of writer space" error, rollback and encode status.
                 // Otherwise, if partial data allowed, save the encode state.
                 // Otherwise roll back. If we have already encoded some chunks, we are done; otherwise encode status.
 
-                if (encodeState.AllowPartialData() && ((err == CHIP_ERROR_BUFFER_TOO_SMALL) || (err == CHIP_ERROR_NO_MEMORY)))
+                if (encodeState.AllowPartialData() && IsOutOfWriterSpaceError(err))
                 {
                     // Encoding is aborted but partial data is allowed, then we don't rollback and save the state for next chunk.
+                    // The expectation is that RetrieveClusterData has already reset attributeReportIBs to a good state (rolled
```

Comment:
Ideally needs review from @yunhanw-google

Reply:
mAllowPartialData is set to true when it needs to tell report engine not to roll back the list items, so the expectation is not right in your comment, the actual expectation is that the state of the buffer is valid to send (i.e. contains no trailing garbage)

Reply:
mAllowPartialData is set to true when it needs to tell report engine not to roll back the list items, so the expectation is not right in your comment, the actual expectation is that the state of the buffer is valid to send (i.e. contains no trailing garbage)

Reply:
mAllowPartialData is set to true when it needs to tell report engine not to roll back the list items, so the expectation is not right in your comment, the actual expectation is that the state of the buffer is valid to send (i.e. contains no trailing garbage)

Reply:
Well, the expectation is that RetrieveClusterData rolled back any partial IBs.  It may have encoded a partial _attribute value_.  I can rephrase to make that clearer.

Reply:
yes, thanks